fe/fe-core/src/main/java/org/apache/doris/nereids/properties/FuncDepsDG.java

```diff
@@ -68,6 +69,12 @@ public DGItem(DGItem dgItem) {
             this.parents = new HashSet<>(dgItem.parents);
             this.children = new HashSet<>(dgItem.children);
         }
+
+        public void replace(Map<Slot, Slot> replaceMap) {
+            this.slots = this.slots.stream()
+                    .map(s -> replaceMap.getOrDefault(s, s))
+                    .collect(Collectors.toSet());
```

Comment:
use `for` loop for better perf. in for loop u could use expected size collection to avoid resize it